plugins/askrene/askrene.c

```diff
@@ -278,7 +278,7 @@ static const char *fmt_flow_full(const tal_t *ctx,
 								     total_delivered,
 								     delay_feefactor)));
 
-	for (int i = tal_count(flow->path) - 1; i >= 0; i--) {
+	for (size_t i = tal_count(flow->path) - 1; i >= 0; i--) {
```

Comment:
You'll get an unintended infinite loop here if i is unsigned.
This applies to every for loop where the stopping condition is `i>=0`.

Reply:
Ah, I didn't think of that. Thanks a lot! I'll just close the pull request. The developers can decide what to do with my comments in the corresponding issue.